src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -877,28 +881,38 @@ private static void ThrowOOM()
                 throw CreateHttpRequestException(new CurlException((int)CURLcode.CURLE_OUT_OF_MEMORY, isMulti: false));
             }
 
-            internal sealed class SendTransferState
+            internal sealed class SendTransferState : IDisposable
```

Comment:
Maybe add a finalizer that returns the buffer to the array pool as well (as a backup to prevent leaks)?